### PR TITLE
std:: int overloads test model

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Submission Checklist

- [x] Run unit tests: `./runTests.py src/test/unit`
- [x] Run cpplint: `make cpplint`
- [x ] Declare copyright holder and open-source license: see below

#### Summary

Adding a model that will test fun(int) and fun(real) for all functions that mirror the std:: cmath functions. 

This is added to test the changes in https://github.com/stan-dev/math/pull/1765

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/test/test-models/good/int_overloads.stan [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  transformed_data_real = log(d_int);
+  transformed_data_real = log(d_real);
+  transformed_data_real = log1p(d_int);
+  transformed_data_real = log1p(d_real);
+  transformed_data_real = log2(d_int);
+  transformed_data_real = log2(d_real);
+  transformed_data_real = log10(d_int);
+  transformed_data_real = log10(d_real);
+
+  transformed_data_real = pow(d_int);
```
Comment (bob-carpenter) [inline src/test/test-models/good/int_overloads.stan]: How does this compile?  `pow` requires two arguments, so we need all permutations.

File: src/test/test-models/good/int_overloads.stan [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  real y_p;
+}
+model {
+  y ~ normal(0,1);
+}
+
+transformed parameters {
+  real transformed_param_real;
+
+  transformed_param_real = abs(d_int);
```
Comment (bob-carpenter) [inline src/test/test-models/good/int_overloads.stan]: These function applications are all already tested in the data section.  These should all be removed and replaced with ones with autodiff variable arguments.  In the case of `pow` and `atan2`, ones that do it in all combinations with `double` and `real`).

**Comment by mcol** [comment]
Also I think that `sqrt`, `tan` and `tanh` appear twice in each block.

File: src/test/test-models/good/int_overloads.stan (hunk 1 above)
Comment (rok-cesnovar) [reply src/test/test-models/good/int_overloads.stan]: Ah, I was too qucik there.

File: src/test/test-models/good/int_overloads.stan (hunk 2 above)
Comment (rok-cesnovar) [reply src/test/test-models/good/int_overloads.stan]: Added.

File: src/test/test-models/good/int_overloads.stan (hunk 1 above)
Comment (rok-cesnovar) [reply src/test/test-models/good/int_overloads.stan]: Fixed.

**Comment by rok-cesnovar** [comment]
Thanks @bob-carpenter and @mcol 

I was a bit sloppy, this should now be cleaned up.

**Pull request merged**